### ISSUE TITLE
Doubly nested input disappears when upgrading to activeadmin 1.0

### ISSUE THREAD

**Issue by alexaltair** [activeadmin/activeadmin#5182]
I have a form that, when maximally simplified, looks like this.
```ruby
ActiveAdmin.register Foo do
  form do |f1|
    f1.inputs 'DeepFoos' do
      f1.has_many :deep_foos do |f2|
        f2.input :name

        f2.inputs "DeeperFoos for #{f2.object.name}" do
          f2.has_many :deeper_foos do |f3|
            f3.input :color
          end
        end
      end
    end
  end
end
```
When I was using activeadmin 0.6.6, this rendered fine, with the double-nested resource. After updating to 1.0.0.pre5, the `DeeperFoos` input part of the form disappeared. There's no exception, and it's not in the html source code of the page. It's simply gone.

Weirdly, if I remove the line that says `f2.input :name`, the `DeeperFoos` input rendered again.

I can't find anything relevant in the [activeadmin upgrade docs](https://github.com/activeadmin/activeadmin/wiki/Upgrading-ActiveAdmin) or the [formtastic upgrade docs](https://github.com/justinfrench/formtastic/wiki/Upgrading-to-Formtastic-3.1) (which, as a dependency, got upgraded from 2.2.1 to 3.1.5).

**Comment by javierjulio** [comment]
@alexaltair I'm not sure `has_many` can be nested like that. @varyonic would you have any idea if that is even or was supported? 😳

**Comment by alexaltair** [comment]
@javierjulio We used it for several months on `activeadmin 0.6.6`, so it did used to work.

**Comment by varyonic** [comment]
The simplest case is tested, see https://github.com/activeadmin/activeadmin/blob/0c342275cd90455ddf6b2787e7a3c3e00bfdaa69/spec/unit/form_builder_spec.rb#L898
However, other scenarios may have been broken by the rewrite.

**Comment by alexaltair** [comment]
Agreed with @varyonic. The difference that I see between my example and the test one is the line `f2.input :name`, that is, the existence of any other input fields on the middle model. If I take that line out, it does render the deepest module.

**Comment by xn** [comment]
Probably related. This only renders the `:value` input in the `has_many` At this point, I think the problem are the compounder refactors going on interfering with other AA plugins. It's a mess.
```form do |f|
  f.semantic_errors *f.object.errors.keys
  f.inputs "Foo" do
    f.input :a_boolean
    f.inputs do
      h3 "w00t"
      f.has_many :bars, allow_destroy: true, new_record: true do |ff|
        ff.input :baz, as: :select, collection: Baz.order(:title)
        ff.input :value
        ff.input :some_select, as: :select, collection: [:zips, :zaps], input_html: { class: "default-select" }
      end
    end
  end
  f.actions
end```

**Comment by varyonic** [comment]
@xn Are you trying to help? If so, can you please offer the simplest scenario that fails/doesn't fail. Are you suggesting having multiple input statements in the inner has_many does not render? If so I tried that [here](https://github.com/activeadmin/activeadmin/pull/5190/files) and it seemed to work fine. I am certainly open to the suggestion there is a bug, but off-hand comments like "compounded refactors... other plugins... It's a mess." don't help at all.

**Comment by xn** [comment]
Yes, this used to work, btw. I used it often. no longer works on master.

**Comment by xn** [comment]
I removed all plugins. This form is not rendering answers at all.

```
  form do |f|
    f.semantic_errors *f.object.errors.keys
    f.inputs "Quiz" do
      f.input :title
      f.input :time
      f.has_many :questions, allow_destroy: true, sortable: :position, new_record: true do |ff|
        ff.input :content
        ff.has_many :answers, allow_destroy: true, sortable: :position, new_record: true do |fff|
          fff.input :content
          fff.input :correct
        end
      end
    end

    f.actions
  end
```

**Comment by varyonic** [comment]
So did you try #5190 or reverting before #5196?

**Comment by xn** [comment]
#5190  works. Reverting works as well. I guess I don't need to open up a PR now. Anything holding up #5190?

**Comment by varyonic** [comment]
@alexaltair Do you still have this issue on master now that #5190 is merged?

**Comment by abdulrahman-ahmad** [comment]
I would like to note that I still have this issue, but I've noticed that by removing the nested `f2.inputs` the nested form rendered, somewhat, normally.
Otherwise, the nested form doesn't even get added to the DOM. 
Hopefully this could help fixing the issue is the source code.
I'll try to look more into it soon.

**Comment by seakelps** [comment]
I am also currently experiencing this issue. Is it expected to be working still or is this functionality no longer supported?

**Comment by varyonic** [comment]
@seakelps As noted above there was a fix in #5190. Which version are you using?

**Comment by joseglego** [comment]
@varyonic 

I'm having a trouble, which I think could be related with this. It's NOT the same. But, it could be related. I tried upgrading to the last version at this time, `master` -- [ef90f33](https://github.com/activeadmin/activeadmin/commit/ef90f33416eae8d06d2f01d5d6c74e26d9c82b89)
and I'm still experimenting this trouble. I tried the example showed and it doesn't work. 

In my case I'm trying to show the inputs isolated by sections. And I have something like this:

```ruby
ActiveAdmin.register Foo do
  form do |f1|
    f1.inputs 'DeepFoos' do
      f1.has_many :deep_foos do |f2|
	    f2.inputs "Core for F2" do
	   	  f2.input :core
        end
	    f2.inputs "Other for F2" do
	   	  f2.input :other
        end
        f2.inputs "Extra for F2" do
	   	  f2.input :extra
        end
      end
    end
  end
end
```
and it only shows _Extra for F2_ (It doesn't show _Core For F2_, or _Other For F2_). So, I thought it was showing the last one. But after found this issue I tried the example showed and it doesn't work, so I tried:

```ruby
ActiveAdmin.register Foo do
  form do |f1|
    f1.inputs 'DeepFoos' do
      f1.has_many :deep_foos do |f2|
     	f2.input :core
	    f2.inputs "Other for F2" do
	   	  f2.input :other
        end
        f2.input :extra
      end
    end
  end
end
```
It shows the `:core` and `:extra` inputs. So, it's not related to show only the last element, and it is ignoring the `f2.inputs` like this issue (for this reason I thought it was related)

**Comment by seakelps** [comment]
@varyonic I'm using 1.3.1.

**Comment by victorlcampos** [comment]
Same here

**Comment by timnolte** [comment]
I needed to upgrade the entire Ruby/Rails stack for a client site and I'm now running into the issue with nested `inputs` and the result is both duplicated form sections as well as completely missing fields. Our application uses heavily the [nested form image upload](https://github.com/activeadmin/activeadmin/wiki/Nested-model-form-image-upload) wiki example. This functionality seems to now be broken and seems related to this same reported issue.

**Comment by dante-wd** [comment]
Currently having the same issue in activeadmin 2.8.1 on rails 6.0.3.2. I have a JSON field that looks like this:

```json
{
  "legal": {
    "street": "xxxxx",
    "number": "yyyyy",
    "floor": "zzzzz"
  },
  "factory": {
    "street": "xxxxx",
    "number": "yyyyy",
    "floor": "zzzzz"
  },
  "samples": {
    "street": "xxxxx",
    "number": "yyyyy",
    "floor": "zzzzz"
  }
}
```

And this is how I'm setting up the form:

```ruby
ActiveAdmin.register Client do
  form do |f|
    f.inputs name: 'Addresses', for: :address do |d|
      d.inputs name: 'Legal', for: :legal do |dl|
        dl.input :street, input_html: {
          value: client.address['legal']['street']
        }
        dl.input :number, input_html: {
          value: client.address['legal']['number']
        }
        dl.input :floor, input_html: {
          value: client.address['legal']['floor']
        }
      end
      d.inputs name: 'Factory', for: :factory do |df|
        df.input :street, input_html: {
          value: client.address['factory']['street']
        }
        df.input :number, input_html: {
          value: client.address['factory']['number']
        }
        df.input :floor, input_html: {
          value: client.address['factory']['floor']
        }
      end
      d.inputs name: 'Samples', for: :samples do |ds|
        ds.input :street, input_html: {
          value: client.address['samples']['street']
        }
        ds.input :number, input_html: {
          value: client.address['samples']['number']
        }
        ds.input :floor, input_html: {
          value: client.address['samples']['floor']
        }
      end
    end

    f.actions
  end
end
```

But only the last section is showing up:

![image](https://user-images.githubusercontent.com/65609794/103430703-7f28cf00-4ba5-11eb-8789-1af00011a855.png)
